docs/key-issues/gpus.md

```diff
@@ -180,3 +204,87 @@ Potential solutions on the PyPI side include:
 - make an environment marker or selector package approach work,
 - improve interoperability with other package managers, in order to be able to
   declare a dependency on a CUDA or ROCm version as externally provided,
+
+
+## Additional Notes on CUDA Compatibility
+
+Compatibility across CUDA versions is a common problem with numerous pitfalls to be aware of.
+There are three primary components of CUDA:
+
+1. The CUDA Toolkit (CTK): This component includes the CUDA runtime library
+   (libcudart.so) along with a range of other libraries and tools including
+   math libraries like cuBLAS. libcudart.so and a few other core headers and
+   libraries compose the bare minimum required to compile CUDA code.
+2. The user-mode driver: This is the libcuda.so library. This library is
+   required to actually run CUDA code.
```

Comment:
In contrast to compiling and linking (from point 1 just above), does this explain more explicitly what it does?

```suggestion
   required to load and run CUDA code.
```